infra/Pos/Communication/Types/Protocol.hs

```diff
@@ -204,13 +206,16 @@ type ListenersWithOut m = ([ListenerSpec m], OutSpecs)
 
 -- | Parser for PeerId. Any base64 string.
 peerIdParser :: P.Parser PeerId
-peerIdParser = fmap PeerId P.base64Url
+peerIdParser = do
+    bytes <- P.base64Url
+    when (BS.length bytes /= 14) $ Monad.fail "PeerId must be exactly 14 bytes"
```

Comment:
@rockbmb it doesn't use MonadFail, it just uses `fail`, which is a reasonable operation for a parser. Using an exception wouldn't be overkill, it would simply be a wrong thing to do.